src/App.tsx

```diff
@@ -1,61 +1,139 @@
+import { useState } from 'react';
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+export function getUserFromId(userId: number): User | null {
+  return usersFromServer.find((person) => person.id === userId) || null;
+}
+
+export const todosWithUser: Todo[] = todosFromServer.map((todo) => ({
+  ...todo,
+  user: getUserFromId(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosWithUser);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [titleError, setTitleError] = useState(false);
+  const [userIdError, setUserIdError] = useState(false);
+  const userList = usersFromServer;
+  let highestTodoId = Math.max(...todos.map(todo => (todo.id)));
+
+  function changeTitle(event: React.ChangeEvent<HTMLInputElement>) {
+    // I opted to do the optional instruction
+    const legalCharacters = 'abcdefghijklmnopqrstuvwxyz1234567890 ' /* 'en' */
```

Comment:
While using it like this can work, a more maintainable and flexible solution would be to use regular expressions to validate and sanitize the input. Usually they will be made by someone else or used from a library so you won't have to worry about it too much.